### PR TITLE
Fix build time processing of Kamelets with bean definitions

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/kamelet.adoc [hunk 1]
```diff
@@ -46,16 +46,18 @@ endif::[]
...
 
-This extension allows to pre-load a set of Kamelets at build time using the `quarkus.camel.kamelet.identifiers` property.
+This extension can preload a set of Kamelets at build time. You must include the names of each Kamelet, in the `quarkus.camel.kamelet.identifiers` configuration property.
 
 [id="extensions-kamelet-usage-using-the-kamelet-catalog"]
 === Using the Kamelet Catalog
 
 A set of pre-made Kamelets can be found on the /camel-kamelets/latest[Kamelet Catalog].
-To use the Kamelet from the catalog you need to copy their yaml definition (that you can find https://github.com/apache/camel-kamelets/[in the camel-kamelet repo]) on your project in the classpath. Alternatively you can add the `camel-kamelets-catalog` artifact to your `pom.xml`:
+To use a Kamelet from the catalog, you need to copy its YAML definition (that you can find https://github.com/apache/camel-kamelets/[in the camel-kamelet repository]) to your project.
```
Comment (aldettinger) [inline docs/modules/ROOT/pages/reference/extensions/kamelet.adoc:58]:
```suggestion
To use a Kamelet from the catalog, you need to copy its YAML definition (that you can find https://github.com/apache/camel-kamelets/[in the camel-kamelets repository]) to your project.
```

File: extensions/kamelet/runtime/src/main/java/org/apache/camel/quarkus/component/kamelet/KameletRecorder.java [hunk 2]
```diff
@@ -63,6 +63,14 @@ public void configure(CamelContext context) {
                         if (definition.getRoute() != null && definition.getRoute().getOutputs() != null) {
                             definition.getRoute().getOutputs().forEach(o -> o.setCamelContext(context));
                         }
+
+                        if (definition.getTemplateBeans() != null) {
+                            definition.getTemplateBeans()
+                                    .stream()
+                                    .filter(bean -> bean.getResource() != null)
```
Comment (aldettinger) [inline extensions/kamelet/runtime/src/main/java/org/apache/camel/quarkus/component/kamelet/KameletRecorder.java:70]: Is `!=` null really needed as handled by instanceof ?

**Review by aldettinger — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/extensions/kamelet.adoc (hunk 1 above)
Comment (jamesnetherton) [reply docs/modules/ROOT/pages/reference/extensions/kamelet.adoc]: Good spot - I have corrected it.

File: extensions/kamelet/runtime/src/main/java/org/apache/camel/quarkus/component/kamelet/KameletRecorder.java (hunk 2 above)
Comment (jamesnetherton) [reply extensions/kamelet/runtime/src/main/java/org/apache/camel/quarkus/component/kamelet/KameletRecorder.java]: Yes, you're right. I removed the null filter.

**Pull request merged**